Trakt/ScheduledTasks/SyncLibraryTask.cs

```diff
@@ -385,140 +465,218 @@ public async Task ExecuteAsync(IProgress<double> progress, CancellationToken can
                 try
                 {
-                    var dataContracts =
-                        await _traktApi.SendLibraryUpdateAsync(
-                                collectedEpisodes,
-                                traktUser,
-                                collected ? EventType.Add : EventType.Remove,
-                                cancellationToken)
-                            .ConfigureAwait(false);
-                    if (dataContracts != null)
+                    var dataContracts = new List<TraktSyncResponse>();
+                    var percentPerRequest = availablePercent / (episodes.Count / 100.0);
+
+                    // Force update trakt.tv if we have more than 100 movies in the queue due to API
+                    while (episodes.Count > 100)
                     {
-                        foreach (var traktSyncResponse in dataContracts)
-                        {
-                            LogTraktResponseDataContract(traktSyncResponse);
-                        }
+                        var episodesToSend = episodes.Take(100).ToList();
+                        dataContracts = await _traktApi.SendEpisodePlaystateUpdates(
+                            episodesToSend,
+                            traktUser,
+                            seen,
+                            cancellationToken)
+                        .ConfigureAwait(false);
+
+                        episodes.RemoveRange(0, 100);
+                        currentProgress += percentPerRequest;
+                        progress.Report(currentProgress);
+
+                        LogTraktResponseDataContract(dataContracts, TraktItemType.episode);
                     }
-                }
-                catch (ArgumentNullException argNullEx)
-                {
-                    _logger.LogError(argNullEx, "ArgumentNullException handled sending episodes to trakt.tv");
+
+                    dataContracts = (await _traktApi.SendEpisodePlaystateUpdates(
+                        episodes,
+                        traktUser,
+                        seen,
+                        cancellationToken)
+                    .ConfigureAwait(false))
+                    .ToList();
+
+                    currentProgress += percentPerRequest;
+                    progress.Report(currentProgress);
+
+                    LogTraktResponseDataContract(dataContracts, TraktItemType.episode);
                 }
                 catch (Exception e)
                 {
-                    _logger.LogError(e, "Exception handled sending episodes to trakt.tv");
+                    _logger.LogError(e, "Error updating episode play states");
                 }
-
-                progress.Report(100);
             }
         }
 
-        private void LogTraktResponseDataContract(TraktSyncResponse dataContract)
+        private void LogTraktResponseDataContract(IEnumerable<TraktSyncResponse> dataContracts, TraktItemType type)
```

Comment:
```suggestion
        private void LogTraktResponseDataContract(IReadOnlyCollection<TraktSyncResponse> dataContracts, TraktItemType type)
```

I prefer to avoid `IEnumerable` when possible